Libraries/process/Sources/DefaultLauncher.cpp

```diff
@@ -80,6 +91,18 @@ launch(Filesystem *filesystem, Context const *context)
         return ext::nullopt;
     } else if (pid == 0) {
         /* Fork succeeded, new process. */
+        /* Setup pipe to parent, redirecting both stdout and stderr */
+        if (pipe_setup_success) {
+            dup2(pfd[1], STDOUT_FILENO);
+            dup2(pfd[1], STDERR_FILENO);
+            close(pfd[0]);
+            close(pfd[1]);
+        } else {
+            /* No parent-child pipe setup, just ignore outputs from child */
+            fclose(stdout);
```

Comment:
`fclose()`ing `stdout` and `stderr` is a no-op here because these will get re-initialized by the child process after the call to `execve()`; without an `execve()`, you'd still have the underlying file descriptors open, so you still have the issue.

The proper way to ignore the output from the child is this:

```
close(STDOUT_FILENO);
close(STDERR_FILENO);
open("/dev/null", O_RDONLY);
open("/dev/null", O_WRONLY);
```

You can add `STDIN_FILENO` to the snippets above if you want to cancel out the input fd as well.